### DISCUSSION TITLE
My project is currently using Node.js v16.19.1, and I can't install @scalar/api-reference package.

### DISCUSSION THREAD

**Discussion by vfa-tanna** [scalar/scalar#1107] [Q&A] [answered]
@scalar/swagger-editor@0.10.15: The engine "node" is incompatible with this module. Expected version ">=18". Got "16.19.1"

**Comment by marclave** [comment]
> @scalar/swagger-editor@0.10.15: The engine "node" is incompatible with this module. Expected version ">=18". Got "16.19.1"
> 
> 

Thanks for reporting this @vfa-tanna ! I will investigate today :) 

Are you able to upgrade your node version or do you have requirements to run the old version :)

**Comment by hanspagel** [comment, marked as answer]
Thanks for reaching out!I don’t know how I feel about this. Node 16 isn’t maintained anymore and could/should be considered insecure. Currently, we support all Node versions, that are still maintained:

* Node 18 (until April 2025)
* Node 20 (until April 2026) and 
* Node 21 (until June 2024)

https://nodejs.org/en/about/previous-releases

If you have any chance to upgrade to Node 18 or 20, that would be a huge benefit (security-wise) for your project.

That said, if you want to (or have to) stick to Node 16, you can still install all packages. It just gives you the following warning:

```
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: '@scalar/swagger-editor@0.10.15',
npm WARN EBADENGINE   required: { node: '>=18' },
npm WARN EBADENGINE   current: { node: 'v16.17.1', npm: '8.15.0' }
npm WARN EBADENGINE }
```

pnpm won’t even bother you. And yarn blocks installation:

```
error @scalar/swagger-parser@0.5.20: The engine "node" is incompatible with this module. Expected version ">=18". Got "16.17.1"
error Found incompatible module.
```

But yarn has an `--ignore-engines` flag to ignore that:

```bash
yarn add @scalar/api-reference --ignore-engines
```

So in all cases, you should be able to install the packages.
